src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -179,6 +179,26 @@ public Exercise findOneLoadParticipations(Long id) {
         return exercise;
     }
 
+    /**
+     * Find exercise by id and load tutorParticipations in this exercise.
+     *
+     * @param id the id of the exercise entity
+     * @return the exercise entity
+     */
+    @Transactional(readOnly = true)
+    public Exercise findOneLoadTutorParticipations(Long id) {
+        log.debug("Request to find Exercise with tutorParticipations loaded: {}", id);
+        Exercise exercise = findOne(id);
+        if(Optional.ofNullable(exercise).isPresent()) {
+            Set<TutorParticipation> tutorParticipations = exercise.getTutorParticipations();
+
+            // Load also the trained example submissions
+            tutorParticipations.forEach(t -> t.getTrainedExampleSubmissions().size());
```

Comment:
This code is not very efficient and leads to multiple database calls. Can we not define a method in the corresponding repository which loads all data at the same time, i.e. the exercise with eager tutor participation and their trained example submissions?

Reply:
Agree, but I refactored all the code which was calling this method and I deleted it altogether :)